tests/ts/arrays_test_complex/arrays_test_complex.fbs

```diff
@@ -0,0 +1,52 @@
+namespace MyGame.Example;
+
+// it appears that the library has already a problem with Enums
+// when generating ts file with '--ts-flat-files' from a fbs.
+// bfbs is fine.
+// workaround is to generate bfbs from fbs first, and then 
```

Comment:
Honestly, I would like to fix the root issue instead of making a workaround. Then we don't need the bfbs.

Reply:
I tried but had not much chance to find the root cause. I believe it needs to be processed seperately.
I'll try to reproduce it to make sure it has no connection with array support.

Reply:
Ah didn't see that this was done as a workaround. Is the enum problem specific to the flat files generation? I have that all reworked in https://github.com/google/flatbuffers/pull/7510

Reply:
@dbaileychess good news is I can no longer reproduce the issue I've mentioned in the fbs file. I must have been doing something wrong. So I removed the bfbs and set the test to use the fbs directly. 92dddf7f121ff251ff169cecef34403c290a59a1

Let me know if you want me to squash everything.

Reply:
> Ah didn't see that this was done as a workaround. Is the enum problem specific to the flat files generation?

Yes .fbs + --ts-flat-files was the combination triggering the issue. (now it doesn't).

>I have that all reworked in #7510

Umm, I'm not sure, has the master got any code from #7510 yet?

Reply:
No 😢

Reply:
I've done several more tests using the various release binaries, build of the current master, this PR and saw no issue with enums at all. I think the last commit resolves this review.

Is it OK to mark this resolved?
Should I squash all three commits in this PR or keep it with three commits?

Reply:
I squash when I merge, so don't worry about that.